packages/core/effects.d.ts

```diff
@@ -477,29 +480,35 @@ export function call<Ctx, Fn extends (this: Ctx, ...args: any[]) => any>(
 export function call<Ctx, Fn extends (this: Ctx, ...args: any[]) => any>(
   ctxAndFn: { context: Ctx; fn: Fn },
   ...args: Parameters<Fn>
-): CallEffect
+): CallEffect<SagaReturnType<Fn>>
 
-export type CallEffect = SimpleEffect<'CALL', CallEffectDescriptor>
+export type CallEffect<RT = any> = SimpleEffect<'CALL', CallEffectDescriptor<RT>>
 
-export interface CallEffectDescriptor {
+export interface CallEffectDescriptor<RT> {
   context: any
-  fn: Function
+  fn: ((...args: any[]) => SagaIterator<RT> | Promise<RT> | RT)
   args: any[]
 }
 
+export type SagaReturnType<S extends Function> =
+  S extends (...args: any[]) => SagaIterator<infer RT> ? RT :
```

Comment:
I think this needs to be something like 
```
ReturnType<S> extends Iterator<any, infer RT, any> ? RT :
```
because ` export type SagaIterator = IterableIterator<StrictEffect>` isn't generic.